nipype/algorithms/confounds.py

```diff
@@ -561,16 +602,25 @@ def _run_interface(self, runtime):
 
         components_file = os.path.join(os.getcwd(),
                                        self.inputs.components_file)
+        components_header = self._make_headers(components.shape[1])
         np.savetxt(
             components_file,
             components,
             fmt=b"%.10f",
             delimiter='\t',
-            header=self._make_headers(components.shape[1]),
+            header='\t'.join(components_header),
             comments='')
+        self._results['components_file'] = os.path.abspath(
+            self.inputs.components_file)
 
-        if self.inputs.pre_filter and self.inputs.save_pre_filter:
-            pre_filter_file = self._list_outputs()['pre_filter_file']
+        save_pre_filter = self.inputs.save_pre_filter
+        if save_pre_filter:
```

Comment:
lines 617-648 can be simpler:

```Python
        if save_pre_filter:
            self._results['pre_filter_file'] = save_pre_filter
            if save_pre_filter is True:
                self._results['pre_filter_file'] = os.path.abspath('pre_filter.tsv')
            <lines 624-648 at this level of indentation>
```

Also replace that `pre_filter_file` with `self._results['pre_filter_file']` in line 643.